3_core_core_analysis/6_explore_secretion_genes.py

```diff
@@ -0,0 +1,126 @@
+
+random.seed(1)
+# -
+
+# ## Load data and metadata
+
+# Input similarity scores and annotations filenames
+# Since the results are similar we only need to look at the scores for one strain type
+pao1_similarity_filename = "pao1_core_similarity_associations_final_spell.tsv"
+
+# Import df
+pao1_similarity = pd.read_csv(pao1_similarity_filename, sep="\t", index_col=0, header=0)
+
+pao1_similarity.head()
+
+# Load KEGG pathway data
+pao1_pathway_filename = "https://raw.githubusercontent.com/greenelab/adage/7a4eda39d360b224268921dc1f2c14b32788ab16/Node_interpretation/pseudomonas_KEGG_terms.txt"
+
+pao1_pathways = annotations.load_format_KEGG(pao1_pathway_filename)
+print(pao1_pathways.shape)
+pao1_pathways.head()
+
+# ## Get genes related to secretion pathways
+
+pao1_pathways.loc[
+    [
+        "KEGG-Module-M00334: Type VI secretion system",
+        "KEGG-Module-M00332: Type III secretion system",
+        "KEGG-Module-M00335: Sec (secretion) system",
+    ]
+]
+
+# Get genes related to pathways
+T6SS_genes = list(pao1_pathways.loc["KEGG-Module-M00334: Type VI secretion system", 2])
+T3SS_genes = list(pao1_pathways.loc["KEGG-Module-M00332: Type III secretion system", 2])
+secretion_genes = list(
+    pao1_pathways.loc["KEGG-Module-M00335: Sec (secretion) system", 2]
+)
+
+# Pull out genes related to T3SS
+T6SS_similarity = pao1_similarity.reindex(T6SS_genes)
+T3SS_similarity = pao1_similarity.reindex(T3SS_genes)
+sec_similarity = pao1_similarity.reindex(secretion_genes)
+
+T6SS_similarity.sort_values(by="Transcriptional similarity across strains")
+
+T3SS_similarity.sort_values(by="Transcriptional similarity across strains")
+
+sec_similarity.sort_values(by="Transcriptional similarity across strains")
+
+# ## Plot
+
+# +
+plt.figure(figsize=(10, 8))
+sns.violinplot(
+    data=pao1_similarity,
+    x="Transcriptional similarity across strains",
+    palette="Blues",
+    inner=None,
+)
+sns.swarmplot(
+    data=T6SS_similarity,
+    x="Transcriptional similarity across strains",
+    color="k",
+    label="T6SS genes",
+    alpha=0.8,
+)
+
+sns.swarmplot(
+    data=T3SS_similarity,
+    x="Transcriptional similarity across strains",
+    color="r",
+    label="T3SS genes",
+    alpha=0.8,
+)
+
+sns.swarmplot(
+    data=sec_similarity,
+    x="Transcriptional similarity across strains",
+    color="yellow",
+    label="secretion system genes",
+    alpha=0.8,
+)
+
+plt.title("Stability of secretion system genes", fontsize=14)
+plt.legend()
+# -
+
+# We hypothesized that most secretion machinery genes would be conserved but that secreted proteins (i.e. effector proteins) would be less conserved. In general, the effector proteins were not included in the KEGG annotations, which is probably why these secretion systems were found to be highly stable.
+#
+# There are some low scoring genes found, especially for the T6SS. Based on a manual inspection, these appear to be _vgrG_ genes that do encode some of the effector proteins.
```

Comment:
Neat analysis! So this is what you expected to find?

Reply:
Yea that was nice to see. I still need to dive more into the genes with a low similarity score